From 86ecea7320bedbde7c7450d2a2932045baba2b08 Mon Sep 17 00:00:00 2001
From: "frank@DESKTOP-2J3C6NR" <fkelly12054@gmail.com>
Date: Tue, 3 Nov 2020 15:55:54 -0500
Subject: [PATCH] Signed-off-by: frank@DESKTOP-2J3C6NR <fkelly12054@gmail.com>
 one letter change

Signed-off-by: frank@DESKTOP-2J3C6NR <fkelly12054@gmail.com>
---
 juicebox.module | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/juicebox.module b/juicebox.module
index ec87e85..824c7fb 100644
--- a/juicebox.module
+++ b/juicebox.module
@@ -31,7 +31,7 @@ function juicebox_theme() {
 * Implements hook_theme_suggestions_HOOK_alter().
 */
 function juicebox_theme_suggestions_field_alter(array &$suggestions, array $variables) {
-  // Juicebox output. Specifically they target a theme-specific selector,
+  // juicebox output. Specifically they target a theme-specific selector,
   // that identifies the field type (e.g. "field-type-image"), with special CSS.
   // This CSS often assumes an inline list of images that have layout/width is
   // being renderd, and apply float or table positioning on the field wrapper.
-- 
2.29.2.windows.1

